### DISCUSSION TITLE
希望能提交一个pr优化表格复原及ocr排序

### DISCUSSION THREAD

**Discussion by Joker1212** [RapidAI/TableStructureRec#20] [General]
1.ocr的识别框明显边界比表格模型的清晰，可以match后得到xmin,ymin,xmax,ymax修正表格识别框
![image](https://github.com/user-attachments/assets/407fbc1e-7d72-4567-b45e-85b6031cf9bf)
2.判断contained逻辑修正为非包含区域面积/整体面积 < 阈值，会比 >= 稳定性更高
![image](https://github.com/user-attachments/assets/05c179de-e6e6-461a-a11e-7ab811b18954)
3.发现有场景logic_point的r_s = 1.5002. r_e = 1.4897,导致四舍五入后 r_e < r_s,需要增加一个阈值先判断是不是同一行
4.发现没有合并同行同列两个识别框，导致数据丢失
5.发现html输出时，完全依赖排序的顺序，没有利用上逻辑行列信息，导致输出表格错位
![image](https://github.com/user-attachments/assets/d932f034-68b0-4f8b-91b3-18a49a10c9d5)
6.ocr的box排序规则修正逻辑可以优化为判断 y 轴的 containes, 同2的判断，比固定10更准确，目前已经遇到反例了
![crop](https://github.com/user-attachments/assets/d010718b-1e6c-4ce2-87fe-2390beda96fc)
![image](https://github.com/user-attachments/assets/011e4c77-04a9-4629-926c-215d34c1913c)
最后，因为最近也在搞文档rag的东西，写了很多包括pdf解析，layout导出了layoutlmv模型的onnx，可以看到https://www.modelscope.cn/models/jockerK/layoutlmv3-onnx
以及正在弄的先分类有线无线表格，再进行识别，非常认同rappid的目标和方案，也用了很多相关项目，希望能跟作者共建rappid生态

**Comment by SWHL** [comment]
我去，第一次见到如此详细的反馈。欢迎提PR来完善这个表格识别哈。
方便的话，可以将微信号发我邮箱（liekkaskono@163.com），交流方便。

**Comment by Athena233** [comment]
> 1.ocr的识别框明显边界比表格模型的清晰，可以match后得到xmin,ymin,xmax,ymax修正表格识别框 ![image](https://private-user-images.githubusercontent.com/19517038/365697319-407fbc1e-7d72-4567-b45e-85b6031cf9bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NjY3ODMsIm5iZiI6MTczNTg2NjQ4MywicGF0aCI6Ii8xOTUxNzAzOC8zNjU2OTczMTktNDA3ZmJjMWUtN2Q3Mi00NTY3LWI0NWUtODViNjAzMWNmOWJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTAzVDAxMDgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiNDhlYmM2M2NiMDljZDY3OGQxMWE1OGQxYTIwMzMwYjliZDJkMWYzZGI5ODU4ZjhkYjBiYTM2MDgwM2NlODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9KR3KjTDT2T59YmR4PD0AJjLg7tr8PCeb9WfhMpZK1g) 2.判断contained逻辑修正为非包含区域面积/整体面积 < 阈值，会比 >= 稳定性更高 ![image](https://private-user-images.githubusercontent.com/19517038/365699676-05c179de-e6e6-461a-a11e-7ab811b18954.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NjY3ODMsIm5iZiI6MTczNTg2NjQ4MywicGF0aCI6Ii8xOTUxNzAzOC8zNjU2OTk2NzYtMDVjMTc5ZGUtZTZlNi00NjFhLWExMWUtN2FiODExYjE4OTU0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTAzVDAxMDgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiYzU0OTg0OTMwNDdmYzk4YmMyYzExNjk4YmM0YWRjZGFjYjFiNTRmMWVjNzVjNjlhOTM2M2Q0YzZmMDc5MDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._sz2D4MM2JYwa45H76QvCf_4mSz_hCR2sblm-utS3c4) 3.发现有场景logic_point的r_s = 1.5002. r_e = 1.4897,导致四舍五入后 r_e < r_s,需要增加一个阈值先判断是不是同一行 4.发现没有合并同行同列两个识别框，导致数据丢失 5.发现html输出时，完全依赖排序的顺序，没有利用上逻辑行列信息，导致输出表格错位 ![image](https://private-user-images.githubusercontent.com/19517038/365698436-d932f034-68b0-4f8b-91b3-18a49a10c9d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NjY3ODMsIm5iZiI6MTczNTg2NjQ4MywicGF0aCI6Ii8xOTUxNzAzOC8zNjU2OTg0MzYtZDkzMmYwMzQtNjhiMC00ZjhiLTkxYjMtMThhNDlhMTBjOWQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTAzVDAxMDgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhY2QwNGU2MWVjYTJlMGU5OTgyYjA0Njg2NWJmZjU0MDIzZGQyZDBjZTk4OGIzZWU2YjMyYmVkM2E4YjA0NzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zL3CWjNBL0HupGiKf78XmLWW2R23RdE9QxqKI7J0yF4) 6.ocr的box排序规则修正逻辑可以优化为判断 y 轴的 containes, 同2的判断，比固定10更准确，目前已经遇到反例了 ![crop](https://private-user-images.githubusercontent.com/19517038/365709153-d010718b-1e6c-4ce2-87fe-2390beda96fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NjY3ODMsIm5iZiI6MTczNTg2NjQ4MywicGF0aCI6Ii8xOTUxNzAzOC8zNjU3MDkxNTMtZDAxMDcxOGItMWU2Yy00Y2UyLTg3ZmUtMjM5MGJlZGE5NmZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTAzVDAxMDgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhOGU2ODE4MGJkM2EyMDAwYTExODMzYTY3MmNkZDRmMDk3YmUwODgwNDYzNWEyMDljODQ3NjdiNjE1MjUwOTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pthFTiHCrQ_w6SI-EyUH2dlNpYJefxyVwf8oR0_TlII) ![image](https://private-user-images.githubusercontent.com/19517038/365709110-011e4c77-04a9-4629-926c-215d34c1913c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU4NjY3ODMsIm5iZiI6MTczNTg2NjQ4MywicGF0aCI6Ii8xOTUxNzAzOC8zNjU3MDkxMTAtMDExZTRjNzctMDRhOS00NjI5LTkyNmMtMjE1ZDM0YzE5MTNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTAzVDAxMDgwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1ZjNiMzMwZmNmYTc1ZDFkYTE1ZWU0ZmE4ZTE2YzA3NWUzMDRhOGU5NTMzZjY1YzA4Y2E1ZmQzZDFmMzhhOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ta9K6scmwcxjzApTeKLONtklZzGbzAloOYgXddsA75k) 最后，因为最近也在搞文档rag的东西，写了很多包括pdf解析，layout导出了layoutlmv模型的onnx，可以看到https://www.modelscope.cn/models/jockerK/layoutlmv3-onnx 以及正在弄的先分类有线无线表格，再进行识别，非常认同rappid的目标和方案，也用了很多相关项目，希望能跟作者共建rappid生态

博主这个提pr了嘛，是不是已经在release里面了呀

**Reply by SWHL** [reply]
已经在最新代码里了
